Translation

marcusblake opened the pull request:
Added feature so that user can set a default language so that messages automatically appear in said language without having to translate each time.

marcusblake committed:
Added simple user-settings.html page to allow user to edit language preferences

marcusblake committed:
Added feature to set language setting and automatically display messages in chosen language

zschreibs reviewed (commented):
Overall looks good, you have to update the user-page.html to avoid a merge conflict (instead of all the addBasicLinkNavigation functions just use the loadNav function (I'm not 100% sure that's what its called) and then make sure to add the user settings page to that function.  Also I find that I'm not able to post messages on this branch, which is an issue i'm having on a few of them I don't know why though...

zschreibs commented:
When I try and change the language in the user settings I get: 500 (Server Error) in the console, and the language does not change

marcusblake committed:
Fixed bug which caused null pointer exception

marcusblake committed:
Added translation functionality for AboutMe, Country, and Name

marcusblake committed:
Added feature to highlight button when user selects a language

marcusblake committed:
Updated branch with changes in master and resolved merge conflict

marcusblake commented:
I updated to fix the bug that you described. Added additional translating functionality which translates the About Me, Name, and Country. Also added small feature to highlight language when user selects a language. Also added functionality to translate text when visiting another user's page.

zschreibs commented on src/main/webapp/user-settings.html:
```diff
@@ -0,0 +1,49 @@
+</head>
+
+
+<body onload="addLoginOrLogoutLinkToNavigation(); addBasicLinkNavigation('stats.html', 'Stats Page'); addBasicLinkNavigation('feed.html', 'Public Feed');addBasicLinkNavigation('charts.html', 'Charts Page');addBasicLinkNavigation('map.html', 'Maps Page');addBasicLinkNavigation('user-map.html', 'User Map');
```
change this whole thing to loadNavBar()

zschreibs commented on src/main/webapp/user-settings.html:
```diff
@@ -0,0 +1,49 @@
+<nav>
+    <ul id="navigation">
+      <li><a href="/">Home</a></li>
+    </ul>
```
add
<li class="dropdown">
        <a href="javascript:void(0)" class="dropbtn">More</a>
        <ul class="dropdown-content" id="dropdown-content"></ul>
      </li>
under the Home li element so the dropdown menu still appears

zschreibs reviewed (commented):
Looks good! Small edits but otherwise awesome!

marcusblake committed:
Updated branch to reflect latest changes in master

zschreibs reviewed (approved):
Looks good!

marcusblake committed:
Merge branch 'master' into translate
